code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -637,6 +640,11 @@ pub mod pallet {
 					share_asset_id,
 					financial_nft_asset_id,
 				} => {
+					// AssetIds must be greater than 0
+					ensure!(!pool_asset.is_zero(), Error::<T>::InvalidAssetId);
```

Comment:
can we use Validated?

Reply:
After talking to the rest of the Parachain finance team this morning, we've decided not to move this to Validated yet.
Currently, Validated does not properly emit errors to the runtime due to an oversight in its design.
Eventually, it would be good to move this, and other checks, to validated but we can't be doing that until we are receiving accurate errors.